### PR TITLE
Adding select

### PR DESCRIPTION
**Pull request by 32bitkid**
`select` works similar to `alts`, however it will return both the key of the channel that was selected as well value from the channel.

Inspired by golang's [reflect.Select](https://golang.org/pkg/reflect/#Select)

### CODE REVIEW

File: test/index.spec.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ test('[csp] methods', t => {
   t.ok(csp.put, 'csp.put should be defined');
   t.ok(csp.take, 'csp.take should be defined');
   t.ok(csp.alts, 'csp.alts should be defined');
+  t.ok(csp.select, 'csp.drain should be defined');
```
Comment (f5io) [inline test/index.spec.js]: should say `csp.select`

File: README.md [hunk 2]
```diff
@@ -100,6 +100,21 @@ put(chan2, 42);
 const msg = await alts(chan1, chan2); // will receive 42
 ```
 
+### `select(Map<*, channel>|Set<channel>|Array<channel>|Object<string, channel>)` -> `Promise`
```
Comment (f5io) [inline README.md]: awesome! could you just bump the number of exposed methods in API section intro.

**Comment by 32bitkid** [comment]
@f5io no problem, man. i just made some minor changes to the way it handles being given a `Set<channel>`; I realized in my sleep that returning an index doesn't make any sense there. If you _do_ have any suggestions to improve the implementation, let me know! Thanks.

**Comment by f5io** [comment]
@32bitkid LGTM!

**Comment by f5io** [comment]
Released as patch `1.0.2` 🎉

**Comment by chriswessels** [comment]
Thanks @32bitkid!

**Pull request merged**